packages/web/src/services/DatabaseServiceWeb.ts

```diff
@@ -78,55 +75,39 @@ export default class DatabaseServiceWeb extends DatabaseService {
         this.database?.detach();
     }
 
-    public async addDataSource(dataSource: Source): Promise<void> {
-        const { name, type, uri } = dataSource;
+    protected async addDataSource(
+        name: string,
+        type: "csv" | "json" | "parquet",
+        uri: string | File
+    ): Promise<void> {
         if (!this.database) {
             throw new Error("Database failed to initialize");
         }
-        if (this.existingDataSources.has(name)) {
-            return;
-        }
-        if (!type || !uri) {
-            throw new DataSourcePreparationError(
-                `Data source type and URI are missing for ${dataSource.name}`,
-                dataSource.name
+
+        if (uri instanceof File) {
+            await this.database.registerFileHandle(
+                name,
+                uri,
+                duckdb.DuckDBDataProtocol.BROWSER_FILEREADER,
+                true
             );
+        } else {
+            const protocol = uri.startsWith("s3")
+                ? duckdb.DuckDBDataProtocol.S3
+                : duckdb.DuckDBDataProtocol.HTTP;
+
+            await this.database.registerFileURL(name, uri, protocol, false);
         }
 
-        this.existingDataSources.add(name);
-        try {
-            if (uri instanceof File) {
-                await this.database.registerFileHandle(
-                    name,
-                    uri,
-                    duckdb.DuckDBDataProtocol.BROWSER_FILEREADER,
-                    true
-                );
-            } else if (typeof uri === "string") {
-                const protocol = uri.startsWith("s3")
-                    ? duckdb.DuckDBDataProtocol.S3
-                    : duckdb.DuckDBDataProtocol.HTTP;
-
-                await this.database.registerFileURL(name, uri, protocol, false);
-            } else {
-                throw new Error(
-                    `URI is of unexpected type, should be File instance or String: ${uri}`
-                );
-            }
-
-            if (type === "parquet") {
-                await this.execute(`CREATE TABLE "${name}" AS FROM parquet_scan('${name}');`);
-            } else if (type === "json") {
-                await this.execute(`CREATE TABLE "${name}" AS FROM read_json_auto('${name}');`);
-            } else {
-                // Default to CSV
-                await this.execute(
-                    `CREATE TABLE "${name}" AS FROM read_csv_auto('${name}', header=true, all_varchar=true);`
-                );
-            }
-        } catch (err) {
-            await this.deleteDataSource(name);
-            throw new DataSourcePreparationError((err as Error).message, name);
+        if (type === "parquet") {
```

Comment:
Could converting the `name` variable directly to a SQL query be vurnrable to probaimatic characters here? I think ";" would break this if its in the name or something like "TABLE" alread in the name.